### PR TITLE
Add LoadMoreButton component

### PR DESCRIPTION
**Pull request by iamgarima**
Along with other updates in `load-more-stories-base.js` to support customisable `offset` and `storiesOnClick`(number of stories to load on clicking load more button)

### CODE REVIEW

**Comment by iamgarima** [comment]
For issue #278 on madrid

File: components/load-more-button.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+function LoadMoreButton(props) {
+    return <button styleName={props.styleName} onClick={props.onLoadMore}>Load more</button>
+}
+
+export { LoadMoreButton };
```
Comment (thisizkp) [inline components/load-more-button.js]: not sure, will this work? Whenever babel encounters `styleName` property, it tries to find the corresponding module which has the respective `styleName` property. In this case it'll be undefined?

File: components/load-more-stories-base.js [hunk 2]
```diff
@@ -48,31 +48,39 @@ export class LoadMoreStoriesManager extends React.Component {
...
       template: this.props.template,
-      loadStories: (pageNumber) => this.loadMoreStories(pageNumber)
+      loadStories: (pageNumber) => this.loadMoreStories(pageNumber),
+      languageDirection: this.props.languageDirection,
+      stories: this.props.data.stories.slice(0, this.props.initialStoryShowCount),
+      noMoreStories: this.props.data.stories.length <= this.props.initialStoryShowCount,
+      storiesOnClick: this.props.storiesOnClick
     }));
   }
 }
```
Comment (iamgarima) [inline components/load-more-stories-base.js]: Use stories.length for offset and remove noMoreStories from the props

File: components/load-more-button.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+function LoadMoreButton(props) {
+    return <button onClick={props.onLoadMore}>Load more</button>
+}
+
+export { LoadMoreButton };
```
Comment (iamgarima) [inline components/load-more-button.js]: Add defaultProps and propTypes here.

Comment (iamgarima) [reply components/load-more-button.js]: Pass stylename and put styles globally s.t. they are available everywhere.

Comment (iamgarima) [reply components/load-more-button.js]: Add configurable button text.

File: components/load-more-stories-base.js (hunk 2 above)
Comment (iamgarima) [reply components/load-more-stories-base.js]: Add default values to the props being passed

File: components/wrap-collection-layout.js [hunk 4]
```diff
@@ -45,8 +47,8 @@ function WrapCollectionComponent(Component) {
       associatedMetadata: associatedMetadata,
     });
 
-    const component = addLoadMore(Component, data, associatedMetadata.load_more, props.collection.slug, data);
-
+    const component = addLoadMore(Component, data, associatedMetadata.enable_load_more_button, props.collection.slug, associatedMetadata.limit_stories, associatedMetadata.load_more_stories_count);
+    
     return [addClientSide, addLazyLoad].reduce((c, f) => f(c, associatedMetadata), component);
   }
 }
```
Comment (iamgarima) [inline components/wrap-collection-layout.js]: Update naming for addLoadMore, addClientSide, addLazyLoad and add a wrapper to addLoadMore for it to follow the pattern

**Pull request merged**